packages/devtools_app/lib/src/release_notes.dart

```diff
@@ -0,0 +1,222 @@
+      right: densePadding -
+          (_ReleaseNotesViewerState.viewerWidth - animation.value),
+      width: _ReleaseNotesViewerState.viewerWidth,
+      child: Card(
+        elevation: defaultElevation,
+        color: theme.scaffoldBackgroundColor,
+        clipBehavior: Clip.hardEdge,
+        shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(defaultBorderRadius),
+          side: BorderSide(
+            color: theme.focusColor,
+          ),
+        ),
+        child: Column(
+          children: [
+            AreaPaneHeader(
+              title: const Text('What\'s new in DevTools?'),
+              needsTopBorder: false,
+              rightActions: [
+                IconButton(
+                  padding: const EdgeInsets.all(0.0),
+                  onPressed: () =>
+                      releaseNotesController.toggleReleaseNotesVisible(false),
+                  icon: const Icon(Icons.close),
+                ),
+              ],
+            ),
+            markdownData == null
+                ? const Text('Stay tuned for updates.')
+                : Expanded(
+                    child: Markdown(
+                      data: markdownData,
+                      onTapLink: (_, href, __) => launchUrl(href, context),
+                    ),
+                  ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class ReleaseNotesController {
+  ReleaseNotesController() {
+    _init();
+  }
+
+  ValueListenable<String> get releaseNotesMarkdown => _releaseNotesMarkdown;
+
+  final _releaseNotesMarkdown = ValueNotifier<String>(null);
+
+  ValueListenable<bool> get releaseNotesVisible => _releaseNotesVisible;
+
+  final _releaseNotesVisible = ValueNotifier<bool>(false);
+
+  void _init() {
+    if (server.isDevToolsServerAvailable && !isEmbedded()) {
+      _maybeFetchReleaseNotes();
+    }
+  }
+
+  void _maybeFetchReleaseNotes() async {
+    final lastReleaseNotesShownVersion =
+        await server.getLastShownReleaseNotesVersion();
+    SemanticVersion previousVersion;
+    if (lastReleaseNotesShownVersion.isEmpty) {
+      previousVersion = SemanticVersion();
+    } else {
+      previousVersion = SemanticVersion.parse(lastReleaseNotesShownVersion);
+    }
+    const devtoolsVersion = devtools.version;
+    final currentVersion = SemanticVersion.parse(devtoolsVersion);
+    if (currentVersion > previousVersion) {
+      try {
+        String releaseNotesMarkdown =
+            await http.read(Uri.parse(_releaseNotesUrl(devtoolsVersion)));
+        // This is a workaround so that the images in release notes will appear.
+        // The {{site.url}} syntax is best practices for the flutter website
+        // repo, where these release notes are hosted, so we are performing this
+        // workaround on our end to ensure the images render properly.
+        releaseNotesMarkdown = releaseNotesMarkdown.replaceAll(
+          '{{site.url}}',
+          'https://docs.flutter.dev',
```

Comment:
Suggestion: Change this and path in `_releaseNotesUrl` to top-level variables

Reply:
done